packages/gatsby/cache-dir/gatsby-browser-entry.js

```diff
@@ -18,9 +18,18 @@ const prefetchPathname = loader.enqueue
 const StaticQueryContext = React.createContext({})
 
 function StaticQueryDataRenderer({ staticQueryData, data, query, render }) {
+  let combinedStaticQueryData = staticQueryData
+
+  if (process.env.GATSBY_EXPERIMENT_LAZY_DEVJS) {
+    combinedStaticQueryData = {
+      ...loader.getStaticQueryResults(),
+      ...staticQueryData,
+    }
+  }
```

Comment:
I think this shouldn't be needed anymore since we merge 2 separate caches (loader and socket.io), but also this is fine to keep for now especially behind the flag, but we can mark this to check this in follow up?

Reply:
yeah that makes sense — so when I remove this, we can just use the context?

Reply:
You should be yes, because https://github.com/gatsbyjs/gatsby/blob/6858f22fc9a64f71756d5cd5122fab99d12749cf/packages/gatsby/cache-dir/query-result-store.js#L94-L131 has the context provider and value is pulled from `loader` on updates (either toward the end of `loadPage` execution or on hot data updates)

But I didn't run it locally without it, so hard to say - so that's why I think it might be ok to keep it for now to minimize amount of changes needed to get this in and streamline afterward if it turns out to not be needed?

Reply:
pushed